src/ray/util/util.h

```diff
@@ -87,6 +106,19 @@ inline std::string AppendToEachLine(const std::string &str,
   return ss.str();
 }
 
+// Returns the TID of the calling thread.
+#ifdef __APPLE__
+inline uint64_t GetTid() {
+  uint64_t tid;
+  RAY_CHECK_EQ(pthread_threadid_np(NULL, &tid), 0);
+  return tid;
+}
+#elif defined(_WIN32)
+inline DWORD GetTid() { return GetCurrentThreadId(); }
+#else
+inline pid_t GetTid() { return syscall(__NR_gettid); }
```

Comment:
I checked `src/ray/thirdparty/dlmalloc.c`, its `GetTid()` impl is `GetCurrentThreadId()` and `pthread_self()`. I think we can do the same here?

Reply:
From `man pthread_self`: "The thread ID returned by pthread_self() is not the same thing as the kernel thread ID returned by a call to gettid(2)."

Thus, it is better to stick with the gettid syscall so that the TID in the output log matches the TID in ps, perf, etc.

Reply:
Got it. 

What's the difference between `syscall(__NR_gettid);` and `gettid()`, should we just call the latter one?